libsolidity/codegen/ContractCompiler.cpp

```diff
@@ -295,6 +302,11 @@ void ContractCompiler::appendFunctionSelector(ContractDefinition const& _contrac
 		appendCalldataUnpacker(functionType->parameterTypes());
 		m_context.appendJumpTo(m_context.functionEntryLabel(functionType->declaration()));
 		m_context << returnTag;
+		m_context.adjustStackOffset(
+			CompilerUtils(m_context).sizeOnStack(functionType->returnParameterTypes()) -
+			CompilerUtils(m_context).sizeOnStack(functionType->parameterTypes()) -
+			1
```

Comment:
I would like to keep it like that. Your suggestion is against the coding style, so we could introduce local variables and make it `adjustStackOffset(ret - params - 1)` but yeah...